modules/kernel/mkosi.clangd

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+set -eu
+
+if [ -z "${MKOSI_CONFIG-}" ]; then
+    exec mkosi \
+        --incremental=strict \
+        --build-sources-ephemeral=no \
+        --format=none \
+        --build-script= \
+        --build-script="$(realpath "$0")" \
+        build
+fi
+
+DISTRIBUTION="$(jq -r .Distribution "$MKOSI_CONFIG")"
+RELEASE="$(jq -r .Release "$MKOSI_CONFIG")"
+ARCH="$(jq -r .Architecture "$MKOSI_CONFIG")"
+
+BUILD_DIRECTORY="$(jq -r .BuildDirectory "$MKOSI_CONFIG")"
+CACHE_DIRECTORY="$(jq -r .CacheDirectory "$MKOSI_CONFIG")"
+# This neglects build sources which have an absolute path as a target
+# but that would make the jq filter more complex
+# and can still added later if someone has a need for this.
+BUILD_SOURCE_MAPPINGS="$(jq -r '[
+    .BuildSources[]
+    | .Source + "/='"$SRCDIR"'/" + .Target
+] | join(",")' "$MKOSI_CONFIG")"
```

Comment:
We should pass this when invoking mkosi with this script as the beginning. You can map the current working directory to `/work/src/$1` if the source dir is passed as the first argument

Reply:
I see no reason why we should simply map all dirs for which we have information available. Only mapping a single one will also be difficult because we do not know which of the `BuildSources` it is. And passing it via CLI does not work because we need to also know the path outside the build sandbox, not only inside. For systemd this works because there is only a single build source usually configured and that is generally `$PWD` on the host

Reply:
@septatrix I would make the assumption that `$PWD` when mkosi.clangd is invoked outside of mkosi is the path outside the sandbox. And the target path inside the sandbox is what's passed in as `$1`

Reply:
But the path inside the sandbox will always be one of `BuildSources`, same for `$PWD` - except for dubious cases where one uses mkosi-kernel as a clangd provider for a different source tree which I think we do not need to support.
Unless I am missing something the current jq filter covers a subset of all mappins which make sense by manually passing them - it just does it automatically for you and for all of them